src/protocol/secureMessages.ts

```diff
@@ -0,0 +1,356 @@
+/**
+ * All messages sent to the Lattice from this SDK will be
+ * "secure messages", of which there are two types:
+ * 
+ * 1. Connect requests are *unencrypted* and serve to establish
+ *    a connection between the SDK Client instance and the target
+ *    Lattice. If the client is already paired to the target Lattice,
+ *    the response will indicate that. If the client has never paired
+ *    with this Lattice, the Lattice will go into "pairing mode" and
+ *    will expect a follow up `finalizePairing` request, which is
+ *    an encrypted request. This will return an ephemeral public key,
+ *    which is used to encrypt the next request.
+ * 2. Encrypted requests are *encrypted* (obviously) and from a Lattice
+ *    protocol perspective they are all constructed the same way:
+ *    create a buffer of `payload` length and fill it with unencrypted
+ *    data, then encrypt the entire payload (not just the data you filled)
+ *    with the ECDH secret formed from the last ephemeral public key.
+ *    The response to this request will contain a new ephemral public
+ *    key, which you will need for the next encrypted request.
+ */
+import {
+  Client
+} from '../index'
+import {
+  ProtocolConstants as Constants,
+  LatticeSecureMsgType,
+  LatticeMsgType,
+  LatticeProtocolVersion,
+  LatticeSecureEncryptedRequestType,
+} from './latticeConstants';
+import {
+  aes256_decrypt,
+  aes256_encrypt,
+  checksum,
+  getP256KeyPairFromPub,
+  randomBytes,
+} from '../util';
+import {
+  getEphemeralId,
+  request,
+} from '../shared/functions';
+import {
+  validateEphemeralPub,
+} from '../shared/validators';
+
+const { msgSizes } = Constants;
+const { secure: szs } = msgSizes;
+
+/**
+ * Build and make a request to connect to a specific Lattice
+ * based on its `deviceId`.
+ * @param client - Instance of (unconnected) client
+ * @param deviceId - Device ID for the target Lattice. Must be in
+ *                   the same `client.baseUrl` domain to be found.
+ * @return {Buffer} - Connection response payload data, which contains
+ *                    information about the connected Lattice.
+ */
+export async function connectSecureRequest(
```

Comment:
Should this be called "request" when it is also serializing?

Reply:
Open to suggestions